### ISSUE TITLE
Development environment setup fails due to migration

### ISSUE THREAD

**Issue by MegaThorx** [gramps-project/Gramps.js#213]
I have follow the steps outlined at https://gramps-project.github.io/web/dev-frontend/setup/ and when I try to run `docker-compose up` it fails with the following error message.

My environment:
OS: Mac OS (M1) and Windows 11 (tried it on both)
Docker: Rancher Desktop with dockerd (moby) (on both machines)

Steps:
* Clone repository
* Run `docker-compose up`

```
gramps-api-dev       | INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
gramps-api-dev       | INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
gramps-api-dev       | INFO  [alembic.runtime.migration] Running upgrade  -> c89728e71264, empty message
gramps-api-dev       | INFO  [alembic.runtime.migration] Running upgrade c89728e71264 -> e5e738d09fa7, Added configuration table
gramps-api-dev       | INFO  [alembic.runtime.migration] Running upgrade e5e738d09fa7 -> e176543c72a8, Add tree column to User
gramps-api-dev       | Traceback (most recent call last):
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
gramps-api-dev       |     self.dialect.do_execute(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/default.py", line 920, in do_execute
gramps-api-dev       |     cursor.execute(statement, parameters)
gramps-api-dev       | sqlite3.OperationalError: duplicate column name: tree
gramps-api-dev       | 
gramps-api-dev       | The above exception was the direct cause of the following exception:
gramps-api-dev       | 
gramps-api-dev       | Traceback (most recent call last):
gramps-api-dev       |   File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
gramps-api-dev       |     return _run_code(code, main_globals, None,
gramps-api-dev       |   File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
gramps-api-dev       |     exec(code, run_globals)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/__main__.py", line 4, in <module>
gramps-api-dev       |     main(prog="alembic")
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/config.py", line 617, in main
gramps-api-dev       |     CommandLine(prog=prog).main(argv=argv)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/config.py", line 611, in main
gramps-api-dev       |     self.run_cmd(cfg, options)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/config.py", line 588, in run_cmd
gramps-api-dev       |     fn(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/command.py", line 385, in upgrade
gramps-api-dev       |     script.run_env()
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/script/base.py", line 582, in run_env
gramps-api-dev       |     util.load_python_file(self.dir, "env.py")
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/util/pyfiles.py", line 94, in load_python_file
gramps-api-dev       |     module = load_module_py(module_id, path)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/util/pyfiles.py", line 110, in load_module_py
gramps-api-dev       |     spec.loader.exec_module(module)  # type: ignore
gramps-api-dev       |   File "<frozen importlib._bootstrap_external>", line 790, in exec_module
gramps-api-dev       |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
gramps-api-dev       |   File "alembic_users/env.py", line 97, in <module>
gramps-api-dev       |     run_migrations_online()
gramps-api-dev       |   File "alembic_users/env.py", line 91, in run_migrations_online
gramps-api-dev       |     context.run_migrations()
gramps-api-dev       |   File "<string>", line 8, in run_migrations
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/runtime/environment.py", line 928, in run_migrations
gramps-api-dev       |     self.get_context().run_migrations(**kw)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/runtime/migration.py", line 628, in run_migrations
gramps-api-dev       |     step.migration_fn(**kw)
gramps-api-dev       |   File "/app/src/alembic_users/versions/e176543c72a8_add_tree_column_to_user.py", line 21, in upgrade
gramps-api-dev       |     op.add_column('users', sa.Column('tree', sa.String(), nullable=True))
gramps-api-dev       |   File "<string>", line 8, in add_column
gramps-api-dev       |   File "<string>", line 3, in add_column
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/operations/ops.py", line 2120, in add_column
gramps-api-dev       |     return operations.invoke(op)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/operations/base.py", line 395, in invoke
gramps-api-dev       |     return fn(self, operation)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/operations/toimpl.py", line 154, in add_column
gramps-api-dev       |     operations.impl.add_column(table_name, column, schema=schema, **kw)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/ddl/impl.py", line 322, in add_column
gramps-api-dev       |     self._exec(base.AddColumn(table_name, column, schema=schema))
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/alembic/ddl/impl.py", line 193, in _exec
gramps-api-dev       |     return conn.execute(  # type: ignore[call-overload]
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1413, in execute
gramps-api-dev       |     return meth(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/sql/ddl.py", line 181, in _execute_on_connection
gramps-api-dev       |     return connection._execute_ddl(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1525, in _execute_ddl
gramps-api-dev       |     ret = self._execute_context(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1841, in _execute_context
gramps-api-dev       |     return self._exec_single_context(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
gramps-api-dev       |     self._handle_dbapi_exception(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 2339, in _handle_dbapi_exception
gramps-api-dev       |     raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
gramps-api-dev       |     self.dialect.do_execute(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/sqlalchemy/engine/default.py", line 920, in do_execute
gramps-api-dev       |     cursor.execute(statement, parameters)
gramps-api-dev       | sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) duplicate column name: tree
gramps-api-dev       | [SQL: ALTER TABLE users ADD COLUMN tree VARCHAR]
gramps-api-dev       | (Background on this error at: https://sqlalche.me/e/20/e3q8)
gramps-api-dev       | Traceback (most recent call last):
gramps-api-dev       |   File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
gramps-api-dev       |     return _run_code(code, main_globals, None,
gramps-api-dev       |   File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
gramps-api-dev       |     exec(code, run_globals)
gramps-api-dev       |   File "/app/src/gramps_webapi/__main__.py", line 199, in <module>
gramps-api-dev       |     cli(
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1130, in __call__
gramps-api-dev       |     return self.main(*args, **kwargs)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1055, in main
gramps-api-dev       |     rv = self.invoke(ctx)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1657, in invoke
gramps-api-dev       |     return _process_result(sub_ctx.command.invoke(sub_ctx))
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1657, in invoke
gramps-api-dev       |     return _process_result(sub_ctx.command.invoke(sub_ctx))
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1404, in invoke
gramps-api-dev       |     return ctx.invoke(self.callback, **ctx.params)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 760, in invoke
gramps-api-dev       |     return __callback(*args, **kwargs)
gramps-api-dev       |   File "/usr/local/lib/python3.9/dist-packages/click/decorators.py", line 26, in new_func
gramps-api-dev       |     return f(get_current_context(), *args, **kwargs)
gramps-api-dev       |   File "/app/src/gramps_webapi/__main__.py", line 119, in migrate_db
gramps-api-dev       |     subprocess.run(cmd, env=env, check=True)
gramps-api-dev       |   File "/usr/lib/python3.9/subprocess.py", line 528, in run
gramps-api-dev       |     raise CalledProcessError(retcode, process.args,
gramps-api-dev       | subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'alembic', 'upgrade', 'head']' returned non-zero exit status 1.
gramps-api-dev exited with code 1
```

**Comment by MegaThorx** [comment]
A workaround solution is to change the version in the users.sqlite


```bash
$ docker volume inspect grampsjs_gramps_users                                                                                                                                           
[
    {
        "CreatedAt": "2023-06-11T12:43:15+02:00",
        "Driver": "local",
        "Labels": {
            "com.docker.compose.project": "grampsjs",
            "com.docker.compose.version": "2.18.1",
            "com.docker.compose.volume": "gramps_users"
        },
        "Mountpoint": "/var/lib/docker/volumes/grampsjs_gramps_users/_data",
        "Name": "grampsjs_gramps_users",
        "Options": null,
        "Scope": "local"
    }
]
```

![](https://ass.merx.dev/gargantuan-forthright-irrawaddydolphin/direct)

---

After updating the version in the users.sqlite it starts up fine, but Gramps Web does not load in firefox due to the following error.

![](https://ass.merx.dev/trifling-warped-xenopus/direct)

But it works fine in chromium

**Comment by DavidMStraub** [comment]
Hi,

I did not have time to look into this bug yet (I am currently using a different development set up), so any help would be appreciated.

> but Gramps Web does not load in firefox due to the following error.

Yes, frontend development only works in Chrome/Chromium at present (see the comment [here](https://gramps-project.github.io/web/dev-frontend/setup/)) since Firefox does not support import assertions.

**Comment by DavidMStraub** [comment]
This should be solved with Web API 1.4.0.
